Select video in see the fishway

fungjj92 opened the pull request:
## Overview

Build out See the Fishway. Videos can be selected and autoplayed in the sidebar with accompanying information.

Connects #36 

### Demo
<img width="1158" alt="Screen Shot 2019-05-14 at 2 54 56 PM" src="https://user-images.githubusercontent.com/10568752/57724223-45195b00-7658-11e9-96c0-0450363355c6.png">

Styling super rough, obviously.

### Notes

I made a followup card to format the datetime, which was way more of a pain than I expected. Couldn't get Moment.js or Date.js working! 👎 #71 


## Testing Instructions
Netlify, visit See the Fishway tab.

Videos should autoplay. Custom controls to come.

fungjj92 committed:
Add See The Fishway video snapshots

Refs #36

fungjj92 committed:
Add highlights buttons and play video on select

Refs #36

fungjj92 committed:
Use themes green for highlight

Refs #36

fungjj92 committed:
Add sidebar fish info

Refs #36

maurizi commented on src/app/src/components/VideoCard.js:
```diff
@@ -0,0 +1,23 @@
+    borderRadius: '100px',
+    border: props.selected
+        ? `${themeGet('colors.greens.1')(props)} 3px solid`
+        : 'none',
```
:tada: nice use of `styled-components` here.

maurizi commented on src/app/src/components/Sidebar.js:
```diff
@@ -1,20 +1,30 @@
+const Sidebar = ({ fish }) => {
+    const { commonName, scientificName, path, notes, timestamp } = fish;
+    const date = new Date(timestamp).toLocaleString();
+    const video = fish && <video src={`${path}.mp4`} autoPlay={true} />;
```
We need to add calls to `pauseTimer()` and `resetTimer()` here similar to in `AboutSlide`: https://github.com/azavea/pwd-fishway/blob/33567962c7826817849a271da692d7c10cfb29a5/src/app/src/components/AboutSlide.js#L84-L86

A nice bonus refactor on top of that you could consider is making a very small `Video` component that wires in these actions and passes the rest of its props to the native `<video>` element, and then using that here and in `AboutSlide`:
```jsx
const Video = (props) => {
   return <video
        onPlay={() => dispatch(pauseTimer())}
        onPause={() => dispatch(resetTimer())}
        onEnded={() => dispatch(resetTimer())} 
       {..props}
    />
};
```

fungjj92 replied:
This is a great suggestion!! Good use of the spread operator too. Tackled in latest commit

maurizi reviewed (changes requested):
Other than the comment re: calling the pause/reset timer for the screensaver this looks great.

fungjj92 committed:
Refactor video component and connect to app timer

Refs #36

fungjj92 commented:
I have to address a bug with the About video's click behavior.

fungjj92 committed:
Create sidebar component for quiz and fish proptypes

The contents of the two sidebars diverged enough that squeezing one component into both scenarios felt wrong

Add prop types for the 3 types of fish objects, to add clarity and type-checking since they can be easily confused

Also fix a bug with the video refactor

Refs #36

maurizi commented on src/app/src/components/AboutSlide.js:
```diff
@@ -48,6 +48,9 @@ export default class AboutSlide extends Component {
     }
+    setVideoRef(ref) {
+        this.videoRef = ref;
+    }
```
What is the purpose of this function? It doesn't seem to be called anywhere...

fungjj92 replied:
Good catch, remnant of something I was trying out.

maurizi reviewed (approved):
Other than the 1 comment I left about what appears to be unused code, this is looking great!

Feel free to merge w/o further review after addressing the comment.

fungjj92 commented:
Thanks for review! Merging.
